pkg/verify/signature.go

```diff
@@ -97,8 +97,7 @@ func VerifySignatureWithArtifactDigest(sigContent SignatureContent, verification
 
 func getSignatureVerifier(verificationContent VerificationContent, tm root.TrustedMaterial) (signature.Verifier, error) {
 	if leafCert := verificationContent.Certificate(); leafCert != nil {
-		// TODO: Inspect certificate's SignatureAlgorithm to determine hash function
-		return signature.LoadVerifier(leafCert.PublicKey, crypto.SHA256)
+		return signature.LoadVerifierFromPublicKey(leafCert.PublicKey, options.WithED25519ph())
```

Comment:
the option to default to use Ed25519 with pre-hash, how would that affect scenarios where say a NIST Curve is used for ECC (e.g. P-256).

Reply:
when the key is ECDSA the ed25519ph option is ignored, because of ECDSASignerVerifier is used. That option is only useful when the key is ED25519. I hope that answer your question.

Reply:
Ok. For RSA the current choice of PKCS#1v1.5 would be used still? So all works as before, except the digest would be chosen based on the key type?

This is good IMHO, but I think a comment explaining this would be helpful, as it may appear confusing otherwise if there is always a PH option?

Reply:
> Ok. For RSA the current choice of PKCS#1v1.5 would be used still? So all works as before, except the digest would be chosen based on the key type?

Correct!

> This is good IMHO, but I think a comment explaining this would be helpful, as it may appear confusing otherwise if there is always a PH option?

Sure, I'll add a comment.